### DISCUSSION TITLE
Plain Text Specification?

### DISCUSSION THREAD

**Discussion by sobotka** [meshula/Nanocolor#3] [Announcements]
Apologies for logging this as an “issue” and not a GitHub discussion, but it seems Discussions aren’t yet available on this repository.

This is *terrific* work folks.

Is there any chance of a text based specification being developed?

[I had authored something that is awfully close to what seems to be guiding ideas for NanoColor](https://docs.google.com/document/d/1EJwLX_4CZEvOsE67e6C0fGQU8GWTvJF5N7aKdObIODM/) back when OpenTimelineIO was pondering colourimetric based ideas.

Not sure it is of use, but I am curious if there’s any plans here?

**Comment by meshula** [comment]
Hi Troy, since you pointed it out, I've activated discussions. Note that there is also a #nanocolor channel in the OCIO Slack (I just invited you to the channel). Your OTIO proposal does indeed it look very similar in spirit. I think what you're proposing there encompasses what nanocolor is meant to solve, but may go further into a larger OCIO-lite project.  

I think a text based specification is very much in line with the project's intent.

**Comment by sobotka** [comment]
Thanks Nick. I posted it because it is an extension of a hack patch I did up a long time ago for metadata within OpenColorIO forever.

Getting a text specification “right” seems crucially important. It seems important to be able to accommodate the entities at a granular level (EG: An EOTF, OOTF, OETF, primaries, achromatic centroid, etc.) while also supplying a low-labour descriptor format for higher order constructs.

I am curious your thoughts on the ability to derive “origin” encodings in such a discussion? EG: Being able to get to the “lowest” post FPGA quantal catch array from some specific colourimetric specification?

You can convert an Issue to the Discussion, as an aside. Might be good to port this one over to the Discussion.

**Comment by meshula** [comment]
Part of the charter of nanocolor is to describe how it interacts with a full pipeline, see slide pasted below from the presentation in this repo. Nanocolor sits in that diagram on the arcs in and out of materialX and Hydra that are tagged with Rec709. So yes, part of the work should be to identify how to bridge nanocolor to a more fully managed world that includes things like eotf and oetf's. (Ignore the file display tag, that seems to be an artifact of GitHub doing something weird)

I do expect that some of what you identify properly belongs in the proposed ocio-lite project, of which nanocolor forms a very specific subset.

<img width="1103" alt="image" src="https://github.com/meshula/Nanocolor/assets/501238/95d04402-bf02-4250-aaa8-5f2c895cbdc3">
